packages/lib/src/artifacts/ZWeb3.ts

```diff
@@ -140,8 +140,12 @@ export default class ZWeb3 {
     return ZWeb3.eth().net.getId();
   }
 
-  public static async getNetworkName(): Promise<string> {
+  public static async getNetworkName(providedNetworkId?: string | number): Promise<string | never> {
     const networkId = await ZWeb3.getNetwork();
+    if (providedNetworkId !== undefined && providedNetworkId !== '*' && Number(networkId) !== Number(providedNetworkId)) {
+      throw Error(`Invalid network id ${providedNetworkId}`);
```

Comment:
I'd rephrase to sth like "Unexpected network ID: requested ${provided} but connected to ${networkId}"